lib/train/extras/os_detect_darwin.rb

```diff
@@ -18,6 +18,9 @@ def detect_darwin
       # TODO: ditto on error
       return false if cmd.stdout.empty?
 
+      uname_m = @backend.run_command('uname -m').stdout.chomp
```

Comment:
Do you think it may makes sense to into a method `detect_arch` for all detection implementations. This would allow us to know where it is located